src/test/java/org/opensearch/knn/index/OpenSearchIT.java

```diff
@@ -900,6 +901,136 @@ public void testKNNIndex_whenBuildVectorGraphThresholdIsProvidedEndToEnd_thenBui
         deleteKNNIndex(indexName);
     }
 
+    public void testKNNIndexSearchFieldsParameter() throws Exception {
+        createKnnIndex(INDEX_NAME, createKnnIndexMapping(Arrays.asList("vector1", "vector2", "vector3"), Arrays.asList(2, 3, 5)));
+        for (int i = 1; i <= 10; i++) {
+            Float[] vector1 = { (float) i, (float) (i + 1) };
+            Float[] vector2 = { (float) i, (float) (i + 1), (float) (i + 2) };
+            Float[] vector3 = { (float) i, (float) (i + 1), (float) (i + 2), (float) (i + 3), (float) (i + 4) };
+            addKnnDoc(
+                INDEX_NAME,
+                Integer.toString(i),
+                Arrays.asList("vector1", "vector2", "vector3"),
+                Arrays.asList(vector1, vector2, vector3)
+            );
+        }
+        int k = 100; // nearest 100 neighbors
+
+        // Create search body, all fields
+        XContentBuilder builder = XContentFactory.jsonBuilder()
+            .startObject()
+            .field("fields", new String[] { "*" })
+            .startObject("query")
+            .startObject("match_all")
```

Comment:
why we are using a match all query here? Since a query is a query and it will solve the purpose, I would recommend having both match_all and a k-NN query to ensure that all cases are covered.